src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -186,280 +214,319 @@ def slice_info_for_literal(self, length, prefix="") -> RawSliceInfo:
-                    # to a non-None value. This info is used elsewhere, as
-                    # literals inside a {% set %} block require special handling
-                    # during the trace.
-                    filtered_trimmed_parts = [
-                        p for p in trimmed_parts if not p.isspace()
-                    ]
-                    if (
-                        len(filtered_trimmed_parts) < 3
-                        or filtered_trimmed_parts[2] != "="
-                    ):
-                        set_idx = len(result)
-                elif (
-                    block_type == "block_end"
-                    and set_idx is not None
-                    and (trimmed_parts[0] in ("endmacro", "endset"))
-                ):
-                    # Exiting a {% set %} block. Clear the indicator variable.
-                    set_idx = None
-                m = regex.search(r"\s+$", raw, regex.MULTILINE | regex.DOTALL)
-                if raw.startswith("-") and m:
+                        tag_contents = self.extract_tag_contents(
+                            str_parts, m_close, m_open, str_buff
+                        )
+
+                    if block_type == "block" and tag_contents:
+                        block_type, block_subtype = self.extract_block_type(
+                            tag_contents[0], block_subtype
+                        )
+                    if block_type == "templated" and tag_contents:
+                        assert m_open and m_close
+                        raw_slice_info = self.track_templated(
+                            m_open, m_close, tag_contents
+                        )
+                self.update_inside_set_or_macro(block_type, tag_contents)
+                m_strip_right = regex.search(
+                    r"\s+$", raw, regex.MULTILINE | regex.DOTALL
+                )
+                if raw.startswith("-") and m_strip_right:
                     # Right whitespace was stripped. Split off the trailing
                     # whitespace into a separate slice. The desired behavior is
-                    # to behave similarly as the left stripping case above.
-                    # Note that the stakes are a bit different, because lex()
-                    # hasn't *omitted* any characters from the strings it
-                    # returns, it has simply grouped them differently than we
-                    # want.
-                    trailing_chars = len(m.group(0))
-                    result.append(
+                    # to behave similarly as the left stripping case. Note that
+                    # the stakes are a bit lower here, because lex() hasn't
+                    # *omitted* any characters from the strings it returns,
+                    # it has simply grouped them differently than we want.
+                    trailing_chars = len(m_strip_right.group(0))
+                    self.raw_sliced.append(
```

Comment:
Perhaps just throw this into a fn for ease of reading its nice to balance 
`handle_left_whitespace_stripping` even if this one is easier

Reply:
I looked at doing that a few different ways. It's definitely doable, but the function ended up taking something like 5 parameters, and I couldn't find a reasonable way to get that down. (I tried to limit the new functions to 3 or fewer parameters.) The logic here is not complex, but it depends on a lot of state, so it seemed like _maybe_ it wasn't improving the code. Thoughts?